utilities/scripts/kcp/setup-kcp-local.sh

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+
+# Checks if a binary is present on the local system
+exit_if_binary_not_installed() {
+  for binary in "$@"; do
+    command -v "$binary" >/dev/null 2>&1 || {
+      echo >&2 "Script requires '$binary' command-line utility to be installed on your local machine. Aborting..."
+      exit 1
+    }
+  done
+}
+
+# Checks if the version is compatible by checking greater than or equal to in the passed arguments
+version_compatibility() { 
+    test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1";
+}
+
+
+# Checks if the go binary exists, and if the go version is suitable to setup KCP env
+check_if_go_v_compatibility() {
+    exit_if_binary_not_installed "go"
+    compatible_v="1.18.0"
+    go_version="$(go version | cut -d " " -f3 | cut -d "o" -f2)"
+    echo "For the KCP local build and setup we need go version equal or greater than: $compatible_v."
+    if version_compatibility $go_version $compatible_v; then
+        echo "$go_version >= $compatible_v, compatibility check success ..."
+    else
+        echo "$go_version is lesser than $compatible_v, exiting ..."
+        exit 1
+    fi
+}
+
+check_if_go_v_compatibility
+
+# install-kcp-binaries will install and build kcp in a temporary enviroment
+# since, we are working on a stable branch current KCP stable version is `release-0.5`
+# make build target of KCP will build the controller of KCP
+KCP_DIR="${KCP_DIR:-}"
+install-build-kcp-binaries() {
+  if [[ -z "${KCP_DIR}" ]]; then
+    exit_if_binary_not_installed "git"
+    KCP_PARENT_DIR="$(mktemp -d -t kcp.XXXXXXXXX)"
+    echo $KCP_PARENT_DIR
+    pushd "${KCP_PARENT_DIR}"
+    git clone https://github.com/kcp-dev/kcp.git
+    KCP_DIR="${KCP_PARENT_DIR}/kcp"
+    pushd kcp
+    KCP_STABLE_BRANCH="release-0.5"
+    git checkout "${KCP_STABLE_BRANCH}"
+    make build
+    popd
+    popd
+    printf "\nThe KCP binaries are installed successfully ...\n"
+  fi
+}
+
+# Initiating the KCP server, the kubectl in this case will be using the KCP's kubeconfig
+kcp-init() {
+  printf "\nInitiating KCP server ...\n"
+  pushd "${TMP_DIR}"
+  pwd
+  "$($KCP_DIR/bin/kcp start)" &> "${TMP_DIR}/kcp.log" &
```

Comment:
Here the KCP process runs in the background on our local machine. For [GITOPSRVCE-158](https://issues.redhat.com/browse/GITOPSRVCE-158), shouldn't the KCP run in an actual physical cluster as a Deployment/Statefulset?

Reply:
From my understanding, the expectation from this story is to set up KCP as the api-server (logical cluster) and we are preassuming that we have logged into an openshift cluster that will be acting as our workload aka physical cluster.
Does that make sense to you?

Reply:
The story also talks about setting up the workspace on a physical cluster i.e to deploy KCP as a Deployment/StatefulSet.  

```
Setup kcp workspace on physical cluster
    * deploy kcp on cluster (in a Deployment/StatefulSet) with the appropriate settings for us
    * deploy a syncer on cluster (in a Deployment/StatefulSet) with appropriate settings for us
```

I might be wrong. @jgwest For this story, do we expect KCP to run on our local machine or as a container in an OCP cluster?